### PR TITLE
Transfer ui based test and build jobs to yaml

### PR DESCRIPTION
**Pull request by atsikham**
Build and publishing tested using separate branch and clone of this pipeline. After merging build should pass.

### CODE REVIEW

**Review by plirglo — CHANGES_REQUESTED** [review]

File: pipeline.yml [hunk 1]
```diff
@@ -1,12 +1,139 @@
...
 pr:
-  - develop
-  - master
+  branches:
+    include:
+      - master
+      - develop
 
 pool:
   name: 'Azure Epiphany cluster - epiphany-build-agents-v2'
```
[the remark this answers is not in the document]
Comment (atsikham) [reply pipeline.yml:13]: Let's check.

**Review by plirglo — APPROVED** [review]

File: pipeline.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (atsikham) [reply pipeline.yml]: [Here](https://abb-epiphany.visualstudio.com/epiphany/_build/results?buildId=3952&view=logs&j=5264e576-3c6f-51f6-f055-fab409685f20) is an example of parameter with an ability to redefine.

File: pipeline.yml [hunk 2]
```diff
@@ -1,12 +1,138 @@
...
+        displayName: 'Read Variables from Makefile'
+        inputs:
+          targetType: 'inline'
+          script: |
+            echo "MAJOR_VERSION: $(MAJOR_VERSION)"
+            echo "REPOSITORY_NAME: $(REPOSITORY_NAME)"
+            echo "IMAGE_NAME: $(IMAGE_NAME)"
+            echo "Build.BuildId: $(Build.BuildId)"
+      - task: CmdLine@2
+        displayName: 'Build and Tag image'
```
Comment (mkyc) [inline pipeline.yml:113]: Can we use [docker task](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/docker?view=azure-devops) here?

**Review by mkyc — CHANGES_REQUESTED** [review]

File: pipeline.yml (hunk 2 above)
Comment (atsikham) [reply pipeline.yml:113]: We can. We spoke with Irek about this, and his suggestion was to do so, not to have dependency on Azure devops. But docker task is used in publishing images as it allows us easy use secret for registry.

File: pipeline.yml [hunk 3]
```diff
@@ -1,12 +1,138 @@
...
+            apt-get install -y wget make ssh curl bc
+            wget https://github.com/mikefarah/yq/releases/download/3.3.4/yq_linux_amd64 -O /usr/bin/yq && \
+              chmod +x /usr/bin/yq
+            az || curl -sL https://aka.ms/InstallAzureCLIDeb | bash
+      - task: GoTool@0
+        displayName: 'Install GO'
+        inputs:
+          version: '1.10'
+          goPath: '/root/go'
+          goBin: '/root/go/bin'
```
[the remark this answers is not in the document]
Comment (atsikham) [reply pipeline.yml]: You're right, there is no reason to do so.

File: pipeline.yml [hunk 4]
```diff
@@ -1,12 +1,138 @@
...
+          targetType: 'inline'
+          script: |
+            READ_MAJOR_VERSION=`grep VERSION\ \?\= Makefile | awk -F ' ?= ' '{print $2}'`
+            READ_REPOSITORY_NAME=`grep USER\ \:\= Makefile | awk -F ' := ' '{print $2}'`
+            READ_IMAGE_NAME=`grep IMAGE\ \:\= Makefile | awk -F ' := ' '{print $2}'`
+            SET_PATH=`echo $GOPATH:$GOBIN:$PATH`
+            echo "##vso[task.setvariable variable=MAJOR_VERSION]$READ_MAJOR_VERSION"
+            echo "##vso[task.setvariable variable=REPOSITORY_NAME]$READ_REPOSITORY_NAME"
+            echo "##vso[task.setvariable variable=IMAGE_NAME]$READ_IMAGE_NAME"
+            echo "##vso[task.setvariable variable=PATH]$SET_PATH"
```
[the remark this answers is not in the document]
Comment (atsikham) [reply pipeline.yml]: Done in a such way.

File: pipeline.yml (hunk 2 above)
Comment (mkyc) [reply pipeline.yml:111]: But we are using Azure DevOps and this is Azure DevOps pipeline definition ....

**Review by plirglo — APPROVED** [review]

File: pipeline.yml (hunk 2 above)
Comment (atsikham) [reply pipeline.yml]: Done in suggested way.

**Review by plirglo — APPROVED** [review]

**Review by plirglo — APPROVED** [review]

**Review by mkyc — APPROVED** [review]

**Pull request merged**